src/mds/Server.cc

```diff
@@ -1386,6 +1387,9 @@ void Server::handle_conf_change(const std::set<std::string>& changed) {
   if (changed.count("mds_forward_all_requests_to_auth")){
     forward_all_requests_to_auth = g_conf().get_val<bool>("mds_forward_all_requests_to_auth");
   }
+  if (changed.count("mds_allow_batched_ops")) {
+    allow_batched_ops = g_conf().get_val<bool>("mds_allow_batched_ops");
+  }
```

Comment:
It's unlikely to actually cause trouble, but I think changing this config could technically allow ops to be reordered incorrectly? So maybe we don't make it changeable live?

Reply:
> I was thinking more
> 
>     1. batching is turned on. Many ops are queued
> 
>     2. batching is turned off
> 
>     3. new op comes in that would have been batched
> 
> 
> I guess at that point the new op gets put on a waitlist, which means it is queued behind the batch head?

Yes

> I was worried that it might be able to grab locks immediately and go first. (Which may also not violate any rules, given they're ops in flight simultaneously, but locking changes make me nervous!)

I do not think this should be possible. I also think if it did that it would not be incorrect.